argo/providers/vai/provider.go

```diff
@@ -415,10 +418,18 @@ func (vaip VAIProvider) DeleteRunSchedule(ctx context.Context, providerConfig VA
 		Name: scheduleId,
 	})
 	if err != nil {
-		return err
+		return handleScheduleNotFound(err)
 	}
 
-	return deleteSchedule.Wait(ctx)
+	return handleScheduleNotFound(deleteSchedule.Wait(ctx))
+}
+
+func handleScheduleNotFound(err error) error {
+	if status.Code(err) == codes.NotFound {
+		return nil
+	} else {
+		return err
+	}
```

Comment:
don't need the else

can just `return err` after the if

Reply:
Done in https://github.com/sky-uk/kfp-operator/pull/336/commits/49525b221496cd1a53026ba84803dd8d55eb56a9